cider/tests/nextgen/compiler/ParserTest.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <google/protobuf/util/json_util.h>
+#include <gtest/gtest.h>
+
+#include "exec/nextgen/parsers/Parser.h"
+#include "exec/plan/parser/SubstraitToRelAlgExecutionUnit.h"
+#include "tests/TestHelpers.h"
+#include "tests/utils/Utils.h"
+
+class NextGenParserTest : public ::testing::Test {
+ public:
+  void executeTest(const std::string& sql) {
+    auto json = RunIsthmus::processSql(sql, create_ddl_);
+    ::substrait::Plan plan;
+    google::protobuf::util::JsonStringToMessage(json, &plan);
+
+    generator::SubstraitToRelAlgExecutionUnit substrait2eu(plan);
+    auto eu = substrait2eu.createRelAlgExecutionUnit();
+
+    auto pipeline = cider::exec::nextgen::parsers::toOpPipeline(eu);
+    EXPECT_EQ(pipeline.size(), 3);
```

Comment:
why the size is hardcode as 3?

Reply:
To test whether ```SourceNode``` was inserted. Just a special test, will changed in follow-up PRs.